HanzariProject/backend/src/main/java/com/hancom/hanzari/controllers/TestImageController.java

```diff
@@ -2,37 +2,89 @@
 
 import java.io.IOException;
 
+import java.io.InputStream;
+import java.net.URLConnection;
+
 import javax.servlet.http.HttpServletResponse;
 
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
 import org.springframework.web.bind.annotation.CrossOrigin;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.jlefebure.spring.boot.minio.MinioException;
 
+import io.minio.GetObjectArgs;
+import io.minio.MinioClient;
+import io.minio.PutObjectArgs;
 
+// CORS 오류 해결하기 위한 어노테이션
 @CrossOrigin(origins = "*", maxAge = 3600)
 @RestController
 @RequestMapping("api/images")
 public class TestImageController {
+  
+	@Autowired
+	private MinioClient minioClient;
+	
+	// 버킷명(Amazon S3 Bucket policy를 지켜야 한다.)
+	String bucketName = "hanzari";
 
-	// MinioClient 클래스 사용하기
-	@GetMapping("/{object}")
-	// 한글과컴퓨터-8층.png.png
-	public void getObject(@PathVariable("object") String object, HttpServletResponse response)
-			throws MinioException, IOException {
-//		InputStream inputStream = minioClient.get(Path.of(object));
-//		InputStreamResource inputStreamResource = new InputStreamResource(inputStream);
-//
-//		// Set the content type and attachment header.
-//		response.addHeader("Content-disposition", "attachment;filename=" + object);
-//		response.setContentType(URLConnection.guessContentTypeFromName(object));
-//
-//		// Copy the stream to the response's output stream.
-//		IOUtils.copy(inputStream, response.getOutputStream());
-//		response.flushBuffer();
+	//이미지 파일 MinIO 서버에 업로드
+	//IOException은 imagePutInputStream의 예외 상황 처리를 위해서이다.
+	@PostMapping
+	public boolean putImageFile(@RequestParam("iamgeFile") MultipartFile file) throws IOException {
+		// 이미지 파일이 잘 저장되었는지 아닌지를 boolean 값으로 알려주기 위한 필드
+		boolean result = false;
+		InputStream imagePutInputStream = file.getInputStream();
+		//System.out.println("getOriginalFilename : "+ file.getOriginalFilename());
+		//Path path = Path.of(file.getOriginalFilename());
+		try {
+			//object 속성이 MinIO 버킷에 저장되는 파일 이름이된다.
+			//stream 속성은 이미지 사이즈 크기 만큼 메모리를 사용하여 파일을 전송한다.	
+			minioClient.putObject(
+				    PutObjectArgs.builder()
+				    .bucket(bucketName)
+				    .object("hanzariFloor")//file.getOriginalFilename())
+				    .stream(imagePutInputStream, imagePutInputStream.available(), -1)
+				    .contentType("multipart/form-data")
```

Comment:
받아온 파일은 "이미지"일 것입니다. 그런데 MINIO에 올릴때는 "multipart/form-data"로 올리고 있네요.
이미 우리는 해당 파일이 Image 인지 안다면 "image/jpeg", "image/png" 가 되어야 하는것 아닌가요?

Reply:
MinIO 서버의 Bucket 안에는 어떠한 데이터 형태도 들어갈 수 있고 그에 따라 좀 더 큰 범위를 생각하여(다른 형태의 파일도 전송된다고 생각하여) 해당 contentType을 multipart/form-data로 설정해두었습니다.
그러나 팀장님 말씀대로 한자리 프로그램에서는 "이미지" 형태의 파일만 전송이 되니 contentType을 image/png로 수정하도록 하겠습니다.

Reply:
multipart/form-data 는 "전송해야 할 데이터를 여러 덩어리로 쪼개서 보내겠다"는 의미입니다.
그러므로, 현재 "1개의 파일"을 MINIO에 넣는것과는 의미가 다릅니다.
그러므로 처음에 본인이 의도인 "다른형태의 파일도 커버"하기 위해서 사용한 "multipart/form-data"도 잘못 사용하고 있는것입니다.

타입을 알 수 있다면, 그에 맞는 MIME-TYPE 을 줘야 합니다.
알 수 없거나, 알 필요가 없을때는 보통 `application/octet-stream`을 사용합니다.

아래 링크 참고하세요~
https://developer.mozilla.org/ko/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

```suggestion
				    .contentType("application/octet-stream")
```

Reply:
감사합니다. 해당 내용도 제가 확실하게 숙지하지 않고 그냥 코드를 작성했습니다.
앞으로는 좀 더 확실하게 확인 후 코드를 작성하도록 하겠습니다.
추가로 저희 프로그램에서는 png 파일만 전송이 되니 contentType을 image/png로 한정해서 설정해도 괜찮을까요?

Reply:
jpg는 업로드 안되게 클라이언트에서 막혀 있나요? 아니면 업로드가 되는데 서버단에서 확인해서 거부하나요?

Reply:
팀장님 해당 문제는 contentType 인자를 file.getContentType()을 통해 전송한 파일의 타입을 알게해주었습니다. 이렇게 코드를 작성해도 괜찮을까요?

Reply:
오프라인으로 설명했듯이, 클라이언트를 믿으면 안된다는것만 숙지한다면 진행해도 좋습니다. ^_^

Reply:
#384 네 알겠습니다. 우선 file.getContentType()을 통해 해당 타입을 읽어오게 했지만 말씀해주신대로 클라이언트에서 명시해준 타입과 실제 타입이 일치하지않아 생기는 문제들이 있을 수 있기에 TODO를 통해 후에 수정하도록 적어두었습니다.